server/api/authMiddleware.js

```diff
@@ -1,22 +1,29 @@
 const isAdmin = (req, res, next) => {
-  console.log('req.user.dataValues', req.user.dataValues)
-  const error = new Error('User is not Admin')
-  error.status = 400
-  if (!req.user || !req.user.dataValues.admin) {
-    next(error)
-  } else {
-    next()
+  if (req.user && req.user.dataValues.admin) next()
+  else {
+    console.log('AUTH-MIDDLEWARE: Request failed @ isAdmin')
+    res.sendStatus(404)
   }
 }
 
 const isAuthenticated = (req, res, next) => {
-  const error = new Error('User is not Authenticated')
-  error.status = 400
-  if (!req.user) {
-    next(error)
-  } else {
+  if (req.user) next()
+  else {
+    console.log('AUTH-MIDDLEWARE: Request failed @ isAuthenticated')
+    res.sendStatus(404)
+  }
+}
+
+const isOwnCart = (req, res, next) => {
+  // admins can do this.
+  if (req.user && req.user.dataValues.admin) next()
+  else if (+req.session.cartId === +req.params.cartId)
+    // but otherwise, you can only do this if it is your cart
     next()
+  else {
+    console.log('AUTH-MIDDLEWARE: Request failed @ isOwnCart')
+    res.sendStatus(404)
   }
 }
 
```

Comment:
Not sure at this point why this is admin-only...

Reply:
It isn't admin only.  It will pass if you are an admin (because admins can do anything).  But it will also pass if you are trying to access your own cart (which is fine).